src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -271,24 +272,25 @@ void handleRequestBody(Buffer requestBody) {
                 context.getTraceId(), context.getSpanId(),
                 context.getProject(), deployment.getName(), requestBody.length());
 
-        context.setRequestBody(requestBody);
-        context.setRequestBodyTimestamp(System.currentTimeMillis());
+        try {
+            ObjectNode json = (ObjectNode) ProxyUtil.MAPPER.readTree(requestBody.getBytes());
+            ObjectNode dialFormat = RequestUtil.convertToDialFormat(json);
```

Comment:
It seems like the conversion function is a good candidate to put it to the function's chain.
Please see `ProxyUtil.processChain(tree, enhancementFunctions)` for more details.

Reply:
The purpose of this feature is to handle new format of the request. We agreed that the best way to do that is to convert new format to the old one. Such conversion should be done at the real beginning of request handling to not modify any processing downstream. 

I think making this as enhancement function in theory may lead to some processing errors in future if something is changed. For example the order of the functions execution or introducing a new function that is called before request transformation.

I can wrap my code as an enhancement function but I don't think It is safe. Any thoughts?

Reply:
Right. It makes sense to run the conversion as the first function anyway. However that might be an unsafe way to move the logic to the functions' chain.